Source/Core/VideoBackends/D3D12/D3DCommandListManager.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "D3DQueuedCommandList.h"
+
+namespace DX12
+{
+
+	// This class provides an abstraction for D3D12 descriptor heaps.
+	class D3DCommandListManager
+	{
+	public:
+
+		D3DCommandListManager(D3D12_COMMAND_LIST_TYPE commandListType, ID3D12Device* device, ID3D12CommandQueue *commandQueue, ID3D12DescriptorHeap** gpuVisibleDescriptorHeaps, unsigned int numGpuVisibleDescriptorHeaps, ID3D12RootSignature* defaultRootSignature);
+
+		void SetInitialCommandListState();
+
+		void GetCommandList(ID3D12GraphicsCommandList **ppCommandList);
+
+		void ExecuteQueuedWork(bool waitForGpuCompletion = false);
+		void ExecuteQueuedWorkAndPresent(IDXGISwapChain* pSwapChain, UINT syncInterval, UINT flags);
+
+		void ProcessQueuedWork();
+
+		void WaitForQueuedWorkToBeRecordedOnCPU();
+		void WaitForQueuedWorkToBeExecutedOnGPU();
+
+		void ClearQueueAndWaitForCompletionOfInflightWork();
+		void DestroyResourceAfterCurrentCommandListExecuted(ID3D12Resource* resource);
+		void ImmediatelyDestroyAllResourcesScheduledForDestruction();
+
+		void AddRef();
+		unsigned int Release();
+
+		bool dirtyVertexBuffer;
+		bool dirtyPso;
+		bool dirtyPSCBV;
+		bool dirtyVSCBV;
+		bool dirtyGSCBV;
+		bool dirtySamplers;
+		UINT currentTopology;
+
+#ifdef USE_D3D12_FREQUENT_EXECUTION
+		UINT _drawsSinceLastExecution;
+		bool _cpuAccessLastFrame;
+		bool _cpuAccessThisFrame;

```

Comment:
Please don't use variable names with prefixed underscores. All variables prefixed with an underscore are reserved for standardized use in the global namespace.